packages/es-components/src/components/containers/drawer/Drawer.js

```diff
@@ -52,48 +40,44 @@ export const Drawer = props => {
     }
 
     onActiveKeysChanged(nextActiveKeys);
-  };
-
-  const getPanels = () =>
-    Children.map(children, (child, index) => {
-      // If there is no key provided, use the panel order as default key
-      const key = child.key || String(index + 1);
-      const { title, titleAside } = child.props;
-      const noPadding = child.props.noPadding || false;
-
-      let isOpen = false;
-      if (isAccordion) {
-        isOpen = activeKeys[0] === key;
-      } else {
-        isOpen = activeKeys.indexOf(key) > -1;
-      }
-
-      const childProps = {
-        key,
-        title,
-        titleAside,
-        noPadding,
-        isOpen,
-        children: child.props.children,
-        onItemClick: () => onItemClick(key),
-        closedIconName,
-        openedIconName
-      };
-
-      return React.cloneElement(child, childProps);
-    });
+  }
 
-  return <StyledDrawer {...other}>{getPanels()}</StyledDrawer>;
-};
+  return (
+    <DrawerContext.Provider value={{ openedIconName, closedIconName }}>
+      <StyledDrawer {...other}>
+        {Children.map(children, (child, index) => {
+          // If there is no key provided, use the panel order as default key
+          const key = child.key || String(index + 1);
+
+          let isOpen = false;
+          if (isAccordion) {
+            isOpen = activeKeys[0] === key;
+          } else {
+            isOpen = activeKeys.indexOf(key) > -1;
+          }
+
+          const childProps = {
+            ...child.props,
+            key,
+            isOpen,
+            onItemClick: () => onItemClick(key)
+          };
+
+          return React.cloneElement(child, childProps);
+        })}
+      </StyledDrawer>
+    </DrawerContext.Provider>
+  );
+}
 
 Drawer.propTypes = {
   /** Set which panels are open */
   activeKeys: PropTypes.oneOfType([
     PropTypes.string,
     PropTypes.arrayOf(PropTypes.string)
   ]),
-  /** Should only contain one or more Drawer.Panel elements */
-  children: drawerPanelPropType,
+  /** Usually only one or more Drawer.Panel elements */
```

Comment:
Sounds like there might be a case where downstream consumers could want to use something other than `Drawer.Panel` elements? If that's the case, do we want to expose `DrawerContext` in the package?

Reply:
I mostly just didn't feel we needed the big function to check for Drawer.Panels when we don't do that for most other components. You *can* put something else in there, and it will render fine (maybe they want to throw a header in there for some reason) but it's pretty obvious it's meant to work with Drawer.Panels. I don't think we need to expose the context.